internal/parser/parser.go

```diff
@@ -1049,6 +1055,70 @@ func (p *parser) repeat(fn func() bool) {
 	}
 }
 
+// open will open a new block. It will expect that the next token
+// is the start token and mark that we expect the end token in the
+// future.
+func (p *parser) open(start, end token.Token) (pos token.Pos, lit string) {
+	pos, lit = p.expect(start)
+	p.blocks[end]++
+	return pos, lit
+}
+
+// more will check if we should continue reading tokens for the
+// current block. This is true when the next token is not EOF and
+// the next token is also not one that would close a block.
+func (p *parser) more() bool {
+	_, tok, _ := p.peek()
+	if tok == token.EOF {
+		return false
+	}
+	return p.blocks[tok] == 0
```

Comment:
Should this be `> 0`?

Reply:
No, the function checks if it should read more. It should only read more if the token is not present in the list of blocks.

Reply:
I get it. For some reason I was thinking that `more` took a terminating token as input and was checking against that. Obviously it is checking that the next token is not a terminating token, i.e. one in the stack